### DISCUSSION TITLE
How can I get the SentMessage dispatched to Messenger?

### DISCUSSION THREAD

**Discussion by TristanPouliquen** [symfony/symfony#43531] [Q&A] [answered]
Hello,

As I understand from the documentation, a `SentMessage` is created with details on the email I sent through the mailer: https://symfony.com/doc/current/mailer.html#debugging-emails

I am using the mailer asynchronously, via Messenger. As I see in the [`MessageHandler`](https://github.com/symfony/mailer/blob/5.3/Messenger/MessageHandler.php), the `__invoke` method returns this `SentMessage`.

Does anyone know if & how I can get this message sent back to my queue so that I can fire an action on this message?

I added it to my routing configuration but it does not seem to be caught in the logs & nothing happens.

**Comment by samgiety** [comment]
I would like to know this too. Since the Mailer is `final` the only option is to make local changes to the `Mailer` class in order to achieve this. 
Does anyone have a better way of doing this? Can I get it if I use the dispatcher? Currently I'm using the `Mailer` as a standalone in my project which does not use `Symfony` otherwise.

I actually found out a way to do it. You can use the `Transport` class directly to send the email. It will then return the `SentMessage`.

e.g.
```php
public function send(): ?SentMessage
{
    $transport = Transport::fromDsn($this->getDns());
    return $transport->send($this->emailMessage);
}
```

**Comment by TristanPouliquen** [comment, marked as answer]
I dont remember what was my problem. By routing the Symfony message in my config, I have been able to have handlers listening on this message correctly.
